### PR TITLE
feat: Support for deploying Nacos on MySQL using the MySQL Operator process

### PR DESCRIPTION
**Pull request by dspo**
#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @your-reviewer

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English | Support for deploying Nacos on MySQL using the MySQL Operator process |
| 🇨🇳 中文    | 支持使用 MySQL Operator 流程部署 Nacos on MySQL |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: build/scripts/docker_image.sh [hunk 1]
```diff
@@ -35,7 +35,7 @@ ARCH="${ARCH:-$(go env GOARCH)}"
 VERSION="$(build/scripts/make-version.sh)"
 IMAGE_TAG="${IMAGE_TAG:-$(build/scripts/make-version.sh tag)}"
 DOCKERFILE_DEFAULT="build/dockerfiles/Dockerfile"
-BASE_DOCKER_IMAGE="registry.erda.cloud/erda/${ARCH}/erda-base:20230130"
+BASE_DOCKER_IMAGE="registry.erda.cloud/erda/multi-arch/erda-base:20230130"
```
Comment (sfwn) [inline build/scripts/docker_image.sh:38]: just remove `multi-arch` this layer, use `registry.erda.cloud/erda/erda-base:20230130`

File: cmd/erda-server/bootstrap.yaml [hunk 2]
```diff
@@ -520,6 +520,8 @@ clickhouse.table.loader@span:
 etcd-election@table-loader:
   _enable: ${QUERY_SPAN_FROM_CLICKHOUSE_ENABLE:true}
   root_path: "/erda/monitor-ck-table-loader-election"
+etcd-mutex:
```
Comment (sfwn) [inline cmd/erda-server/bootstrap.yaml:523]: etcd-mutex@msp-addon-deploymentt

**Review by sfwn — CHANGES_REQUESTED** [review]

File: build/scripts/docker_image.sh (hunk 1 above)
Comment (dspo) [reply build/scripts/docker_image.sh]: 已修改

**Review by sixther-dc — APPROVED** [review]
/approve

**Review by sixther-dc — DISMISSED** [review]
/approve

File: internal/apps/msp/resource/deploy/handlers/mysql/mysql_test.go [hunk 3]
```diff
@@ -26,3 +32,20 @@ func TestTryReadFile(t *testing.T) {
...
+	var sg apistructs.ServiceGroup
+	if err := json.Unmarshal([]byte(sgStr), &sg); err != nil {
+		t.Errorf("failure to yaml.Unmarshal: %v", err)
+	}
+	t.Logf("%+v", sg)
+	dtoMap := ParseResp2MySQLDtoMap(ins, &sg)
+	data, err := json.MarshalIndent(dtoMap, "", "  ")
```
Comment (sfwn) [inline internal/apps/msp/resource/deploy/handlers/mysql/mysql_test.go:46]: you'd better to check map items

File: pkg/strutil/json.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package strutil
+
+import "encoding/json"
+
+func MustString(v any) string {
```
Comment (sfwn) [inline pkg/strutil/json.go:19]: bad name. `MustXXX` will be panic if error happened.

Comment (sfwn) [reply pkg/strutil/json.go:19]: the name should be `TryGetJsonString`

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/apps/msp/resource/deploy/coordinator/coordinator.go [hunk 5]
```diff
@@ -78,23 +88,25 @@ func (p *provider) Deploy(req handlers.ResourceDeployRequest) (*handlers.Resourc
 		}
 	}()
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "failure to defaultHandler.GetResourceInfo for %s:%s/%s", req.Engine, req.Options["version"], req.Az)
```
[the remark this answers is not in the document]
Comment (dspo) [reply internal/apps/msp/resource/deploy/coordinator/coordinator.go]: resolved

File: internal/apps/msp/resource/deploy/coordinator/coordinator.go [hunk 6]
```diff
@@ -62,6 +67,11 @@ func (p *provider) CheckIfNeedRealDeploy(req handlers.ResourceDeployRequest) (bo
 }
 
 func (p *provider) Deploy(req handlers.ResourceDeployRequest) (*handlers.ResourceDeployResult, error) {
+	return p.mutexDeploy(p.deploy, req)
+}
+
+// deploy 是部署 addon 的主要业务逻辑
```
[the remark this answers is not in the document]
Comment (dspo) [reply internal/apps/msp/resource/deploy/coordinator/coordinator.go]: resolved

File: internal/apps/msp/resource/deploy/handlers/mysql/mysql_test.go [hunk 7]
```diff
@@ -26,3 +32,20 @@ func TestTryReadFile(t *testing.T) {
...
...d4","ADDON_NODE_ID":"ae72b1255add94a63a244809f4fd64381","ADDON_TYPE":"qa4bb396ca4b3474ca1d953e6f98aa7d4","DICE_ADDON":"qa4bb396ca4b3474ca1d953e6f98aa7d4","DICE_ADDON_TYPE":"mysql","LOCATION-CLUSTER-SERVICE":"","MYSQL_ROOT_PASSWORD":"444c6564e96b420c8f2cf933838f0700","MYSQL_VERSION":"8.0","SERVICE_TYPE":"ADDONS"},"labels":{"ADDON_GROUP_ID":"mysql"},"selectors":null,"volumes":[{"volumeID":"0","volumePath":"","volumeTp":"nas","storage":20,"containerPath":"/var/backup/mysql","scVolume":{"type":"DICE-NAS","storageClassName":"dice-local-volume","size":20,"targetPath":"/var/backup/mysql","snapshot":{}}},{"volumeID":"1","volumePath":"","volumeTp":"nas","storage":20,"containerPath":"/var/lib/mysql","scVolume":{"type":"DICE-NAS","storageClassName":"dice-local-volume","size":20,"targetPath":"/var/lib/mysql","snapshot":{}}}],"healthCheck":null,"health_check":{"exec":{"cmd":"mysql -uroot -p444c6564e96b420c8f2cf933838f0700  -e 'select 1'"}},"traffic_security":{},"status":"Healthy","reason":"","unScheduledReasons":{},"desiredReplicas":0,"readyReplicas":0}],"serviceDiscoveryKind":"","projectNamespace":"","status":"Healthy","reason":"","unScheduledReasons":{},"desiredReplicas":0,"readyReplicas":0}`
```
[the remark this answers is not in the document]
Comment (dspo) [reply internal/apps/msp/resource/deploy/handlers/mysql/mysql_test.go]: resolved

File: internal/apps/msp/resource/deploy/handlers/mysql/mysql_test.go (hunk 3 above)
Comment (dspo) [reply internal/apps/msp/resource/deploy/handlers/mysql/mysql_test.go:46]: resolved

File: pkg/strutil/json.go (hunk 4 above)
Comment (dspo) [reply pkg/strutil/json.go]: resolved

**Review by sfwn — APPROVED** [review]

**Pull request merged**